epub_translator/tools/xml.py

```
@@ -0,0 +1,35 @@
+        for i, child in enumerate(elem):
+            indent(child, level + 1)
+            if not child.tail or not child.tail.strip():
+                if i == len(elem) - 1:
+                    child.tail = "\n" + indent_str
+                else:
+                    child.tail = "\n" + next_indent_str
+    return elem
+
+
+def iter_with_stack(element: Element) -> Generator[tuple[list[Element], Element], None, None]:
+    """中序遍历：yield parent_path, element"""
```

Ruff (0.14.10)

Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)